ghproxy/ghmetrics/ghpaths.go

```diff
@@ -0,0 +1,210 @@
+package ghmetrics
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	reposRegex          = regexp.MustCompile(`^/repos/(?P<owner>[^/]*)/(?P<repo>[^/]*)(?P<rest>/.*)$`)
+	varAndConstantRegex = regexp.MustCompile(`^/(?P<var>[^/]*)(?P<path>/.*)?$`)
+	constantAndVarRegex = regexp.MustCompile(`^/(?P<path>[^/]*)(?P<var>/.*)?$`)
+)
+
+func getSimplifiedPath(path string) string {
+	fragment := getFirstFragment(path)
+	switch fragment {
+	case "repos":
+		// /:owner/:repo
+		return handleRepos(path)
+	case "user":
+		// do some more
+		return path
+	case "users":
+		// do some more
+		return path
+	case "orgs":
+		// do some more
+		return path
+	case "issues":
+		// do some more
+		return path
+	case "search":
+		// do some more
+		return path
+	case "gists":
+		// do some more
+		return path
+	case "notifications":
+		// do some more
+		return path
+	case "repositories", "emojis", "events", "feeds", "hub", "rate_limits", "teams", "licenses":
+		return path
+	default:
+		logrus.WithField("path", path).Info("Path not handled")
+		return path
+	}
+}
+
+// getFirstFragment returns the first fragment of a path, of a
+// `*url.URL.Path`. e.g. `/repos/kubernetes/test-infra/` will return `repos`
+func getFirstFragment(path string) string {
+	re := regexp.MustCompile(`^/(?P<path>[^/]*).*$`)
+	fragment := re.FindStringSubmatch(path)
+	if len(fragment) < 2 {
+		return path
+	}
+	return fragment[1]
+}
+
+func handleRepos(path string) string {
+	match := reposRegex.FindStringSubmatch(path)
+	result := make(map[string]string)
+	for i, name := range reposRegex.SubexpNames() {
+		if i != 0 && name != "" && i <= len(match) { // skip first and empty
+			if match[i] != "" {
+				result[name] = match[i]
+			}
+		}
+	}
+	if result["owner"] == "" || result["repo"] == "" {
+		logrus.WithField("path", path).Info("Not handling /repos/.. path correctly")
+		return "/repos"
+	}
+	rest := result["rest"]
+	sanitizedPath := fmt.Sprintf("/repos/%s/%s", ":owner", ":repo")
+	if rest == "" || rest == "/" {
+		return sanitizedPath
+	}
+	switch getFirstFragment(rest) {
+	case "issues":
+		return fmt.Sprintf("%s%s", sanitizedPath, handlePrefixedVarAndConstant("/issues", rest))
+	case "keys":
```

Comment:
Could you name the var you're using in the `switch` and collapse all of these into one case?

```go
switch fragment := switch getFirstFragment(rest); fragment {
case "keys", "labels", "milestones", "pulls", ... :
    return fmt.Sprintf("%s/%s%s", sanitizedPath, fragment, handleConstantAndVar(rest))
}
```